### ISSUE TITLE
Log files are stored as ${sys:druid.node.type}.log 

### ISSUE THREAD

**Issue by Subhashini2610** [apache/druid#15967]
I am using a k8s based deployment of Druid and see that each node creates log files that are named as 
```${sys:druid.node.type}.log```

The variable is not getting resolved to the right node type. The log files are stored as ```${sys:druid.node.type}.log``` or ```${sys:druid.node.type}-<date>.log```

### Affected Version

Druid version: 28.0.0

### Description

Due to the above problem, I am not able to index the logs appropriately. Am I missing something here?

**Comment by Subhashini2610** [comment]
Please help here

**Comment by abhishekagarwal87** [comment]
when you log into the pod, can you check the full command line that was used to start the druid process?

**Comment by Subhashini2610** [comment]
@abhishekagarwal87 How do I check that? The logs of the broker only contain the following:
```
2024-02-26T14:50:22+00:00 startup service broker
Setting druid.host=10.0.2.188 in /tmp/conf/druid/cluster/query/broker/runtime.properties
Setting druid.auth.authenticator.BasicMetadataAuthenticator.skipOnFailure=false in /tmp/conf/druid/cluster/query/broker/runtime.properties
Setting druid.broker.http.numThreads=15 in /tmp/conf/druid/cluster/query/broker/runtime.properties
Setting druid.indexer.logs.type=file in /tmp/conf/druid/cluster/query/broker/runtime.properties
Setting druid.emitter.statsd.port=8125 in /tmp/conf/druid/cluster/query/broker/runtime.properties
...
...
...
```
Also, the below is the broker container spec in my helm chart.

<img width="575" alt="Screenshot 2024-02-26 at 9 14 06 PM" src="https://github.com/apache/druid/assets/54573581/8b574d2e-9953-4054-a64c-6a57be8f7e73">

All other nodes also have similar spec, except the args, which refer to the particular node type like coordinator, historical etc.

**Comment by fectrain** [comment]
Hi, I met the similar issue, actually I'm using Helm-druid to manage the K8s deployment.
Usually the different services are deployed in different pods, So I'll just list out all pods(`kubectl get pods -n <namespace>`) and then ssh to the pod with specific pod_name, by this way I'm  able to identify the which service/pod I'm accessing, even though the name of log_file looks like `${sys:druid.node.type}.log`

Btw, what do you use to manage the K8s cluster deployment, also using Helm or [druid-operator](https://github.com/datainfrahq/druid-operator)?

**Comment by fectrain** [comment]
Hi, the issue is because `druid.node.type` not been specified when starting up, whose value is supposed to be the service type(e.g. broker/historical, etc..),  have just made some fix right here

**Comment by Subhashini2610** [comment]
@fectrain Thanks a lot for the fix!!! 

When can I see this in the next release? I somehow do not see this in 30.0.0. Please help.

**Comment by Subhashini2610** [comment]
I do not see the fix in 30.0.0. Please help.
